src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -223,7 +223,7 @@ object Task {
 
     override lazy val instanceId: Instance.Id = Instance.Id(runSpecId, Instance.PrefixMarathon, uuid)
 
-    override val reservationId = idString
+    override val reservationId = Reservation.LegacyId(runSpecId, separator, uuid)
```

Comment:
I still don't quite understand why reservation id must be a property on task id... It just doesn't make sense to me especially since most of the tasks are not even resident... can we just drop this and call Reservation.Id directly?

Reply:
> It just doesn't make sense to me especially since most of the tasks are not even resident

You are correct. This will go away with MARATHON-8517. I did not want to change too much in this diff.